apiserver/storage/storage.go

```diff
@@ -14,55 +14,61 @@ import (
 	"github.com/juju/juju/apiserver/common/storagecommon"
 	"github.com/juju/juju/apiserver/facade"
 	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/environs"
 	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
 	"github.com/juju/juju/status"
 	"github.com/juju/juju/storage"
 	"github.com/juju/juju/storage/poolmanager"
-	"github.com/juju/juju/storage/provider/registry"
 )
 
 func init() {
-	common.RegisterStandardFacade("Storage", 2, NewAPI)
+	common.RegisterStandardFacade("Storage", 2, newAPI)
 }
 
 // API implements the storage interface and is the concrete
 // implementation of the api end point.
 type API struct {
 	storage     storageAccess
+	registry    storage.ProviderRegistry
 	poolManager poolmanager.PoolManager
 	authorizer  facade.Authorizer
 }
 
-// createAPI returns a new storage API facade.
-func createAPI(
+func newAPI(
```

Comment:
Just an FYI - @fwereade and I have been putting adapter functions like this - and the RegisterStandardFacade call - in shim.go.